Specs/DataSources/ModelVisualizerSpec.js

```diff
@@ -372,36 +373,39 @@ describe(
     });
 
     it("Computes bounding sphere with height reference", function () {
-      const time = JulianDate.now();
-      const testObject = entityCollection.getOrCreateEntity("test");
-      const model = new ModelGraphics({
-        heightReference: HeightReference.CLAMP_TO_GROUND,
-      });
-      testObject.model = model;
-
-      testObject.position = new ConstantProperty(
-        new Cartesian3(5678, 1234, 1101112)
-      );
-      model.uri = new ConstantProperty(boxUrl);
-      visualizer.update(time);
-
-      const modelPrimitive = scene.primitives.get(0);
-      const result = new BoundingSphere();
-      let state = visualizer.getBoundingSphere(testObject, result);
-      expect(state).toBe(BoundingSphereState.PENDING);
-
-      return pollToPromise(function () {
-        scene.render();
-        state = visualizer.getBoundingSphere(testObject, result);
-        return state !== BoundingSphereState.PENDING;
-      }).then(function () {
-        expect(state).toBe(BoundingSphereState.DONE);
-        expect(modelPrimitive._clampedModelMatrix).toBeDefined();
-        const expected = BoundingSphere.clone(
-          modelPrimitive.boundingSphere,
-          new BoundingSphere()
+      scene.globe.terrainProvider = createWorldTerrain();
+
+      return scene.globe.terrainProvider.readyPromise.then(() => {
```

Comment:
Why do we need to await this in the test?   What happens if this is still pending in production when `getBoundingSphere` is called?

Reply:
Since `sampleTerrainMostDetailed` already awaits the terrain provider's `readyPromise`, I suppose this is not needed in the spec.